src/utils/apiRequests/api.ts

```diff
@@ -1,18 +1,59 @@
+import type { ImpersonationData } from '../../features/user/Settings/ImpersonateUser/ImpersonateUserForm';
+
 import { getQueryString } from './getQueryString';
 import getsessionId from './getSessionId';
 import { APIError, ClientError } from '@planet-sdk/common';
 import { validateToken } from './validateToken';
-import { ImpersonationData } from '../../features/user/Settings/ImpersonateUser/ImpersonateUserForm';
 import { setHeaderForImpersonation } from './setHeader';
 
 const INVALID_TOKEN_STATUS_CODE = 498;
 
+interface BaseRequestOptions {
+  tenant?: string | undefined;
+  url: string;
+}
+
+interface GetAccountInfo {
+  tenant: string | undefined;
+  token: string | null;
+  impersonationData?: ImpersonationData;
+}
+
+interface GetAuthRequestOptions extends BaseRequestOptions {
+  token: string | null;
+  logoutUser: (value?: string | undefined) => void;
+  header?: Record<string, string> | null;
+  queryParams?: { [key: string]: string };
+  version?: string;
+}
+interface PostAuthRequestOptions extends BaseRequestOptions {
+  data: any;
+  token: string | null;
+  logoutUser: (value?: string | undefined) => void;
+  headers?: Record<string, string>;
+}
+interface DeleteAuthRequestOptions extends BaseRequestOptions {
+  token: string | null;
+  logoutUser: (value?: string | undefined) => void;
+}
+interface PutAuthRequestOptions extends BaseRequestOptions {
+  data?: any;
+  token: string | null;
+  logoutUser: (value?: string | undefined) => void;
+}
```

Comment:
For readability, could you follow a similar order for properties in all these interfaces?